### PR TITLE
Random renames

### PR DESCRIPTION
**Pull request by liach**
See #3125

A few renames should encourage modders to use the right types of random.

### CODE REVIEW

File: mappings/net/minecraft/util/math/random/AsyncRandom.mapping [hunk 1]
```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/class_7384 net/minecraft/util/math/random/BlockingSimpleRandom
+CLASS net/minecraft/class_7384 net/minecraft/util/math/random/AsyncRandom
+	COMMENT A random that can be shared by multiple threads safely.
```
Comment (apple502j) [inline mappings/net/minecraft/util/math/random/AsyncRandom.mapping:2]: Is this really async, or just "async-safe"?

**Review by apple502j — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/math/random/AsyncRandom.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/math/random/AsyncRandom.mapping:2]: This probably needs to be `ThreadSafeRandom` then

**Comment by haykam821** [comment]
Would renaming `nextPredictable` to `nextTriangular` be in scope for this pull request?

**Comment by liach** [comment]
Hmm, does that have a triangular density function?

**Comment by apple502j** [comment]
@liach Yep.

image

**Comment by enbrain** [comment]
Yes, `nextPredictable` has a triangular PDF. Let `g(x)` be the PDF of the uniform distribution `U(0, 1)`. The PDF of `nextPredictable(b, v)` is `f(x) = (1/v)∫[-∞,∞]g((x-b)/v+y)g(y)dy`.

* For `x < b - v`, `f(x) = 0`.
* For `b - v <= x < b`, `f(x) = (1/v)∫[0,1+(x-b)/v]g((x-b)/v+y)g(y)dy = (1/v)(1+(x-b)/v)`.
* For `b <= x <= b + v`, `f(x) = (1/v)∫[0,1-(x-b)/v]g((x-b)/v+y)g(y)dy = (1/v)(1-(x-b)/v)`.
* For `b + v < x`, `f(x) = 0`.

It's an isosceles triangle and matches the PDF in https://en.wikipedia.org/wiki/Triangular_distribution#Symmetric_triangular_distribution.

**Comment by Shnupbups** [comment]
I support a rename of `nextPredictable` - I only named it that due to it being referred to as such in the snapshot blogpost, but a more accurate name would be appreciated I'm sure.

**Review by apple502j — APPROVED** [review]
Let's rename nextPredictable in another PR. This definitely has to be merged before pre-release stage begins, because this will be a quite impactful rename after that.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**